server/Arcadia.Assistant.UserPreferences/UserPreferencesActor.cs

```diff
@@ -0,0 +1,63 @@
+﻿namespace Arcadia.Assistant.UserPreferences
+{
+    using System;
+    using System.Collections.Generic;
+
+    using Akka.Actor;
+    using Akka.Persistence;
+
+    public class UserPreferencesActor : UntypedPersistentActor
+    {
+        private readonly Dictionary<string, UserPreferences> userPreferencesById = new Dictionary<string, UserPreferences>();
+
+        public override string PersistenceId => "user-preferences";
+
+        protected override void OnCommand(object message)
+        {
+            switch (message)
+            {
+                case GetUserPreferencesMessage msg:
+                    var userPreferences = this.userPreferencesById.ContainsKey(msg.UserId)
+                        ? this.userPreferencesById[msg.UserId]
+                        : null;
+                    this.Sender.Tell(new GetUserPreferencesMessage.Response(userPreferences));
+                    break;
+
+                case SaveUserPreferencesMessage msg:
+                    try
+                    {
+                        this.Persist(msg, m =>
+                        {
+                            this.SaveUserPreferences(msg.UserId, msg.UserPreferences);
+                            this.Sender.Tell(new SaveUserPreferencesMessage.Success());
```

Comment:
you sure `this.Sender` will be the same caller? Probably better to have a closure on that?

Reply:
Why `this` should be changed? Moreover, the same works in `CalendarEventsStorageBase`.